### PR TITLE
Implement archive signing in package-registry publish subcommand

### PR DESCRIPTION
**Pull request by yim-lee**

### CODE REVIEW

**Comment by yim-lee** [comment]
@swift-ci please smoke test

File: Sources/PackageRegistryTool/PackageRegistryTool+Publish.swift [hunk 1]
```diff
@@ -255,11 +262,54 @@ extension SwiftPackageRegistryTool {
...
         func sign(
+            packageIdentity: PackageIdentity,
+            packageVersion: Version,
             archivePath: AbsolutePath,
             configuration: PublishConfiguration.Signing,
+            workingDirectory: AbsolutePath,
             observabilityScope: ObservabilityScope
         ) throws -> Data {
-            fatalError("not implemented")
+            try _unsafe_wait {
```
Comment (yim-lee) [inline Sources/PackageRegistryTool/PackageRegistryTool+Publish.swift:272]: Not sure if this is a good idea

File: Sources/PackageSigning/SignatureProvider.swift [hunk 2]
```diff
@@ -113,6 +112,27 @@ public enum SignatureFormat: String {
             return CMSSignatureProvider(format: self)
         }
     }
+
+    public func privateKey(derRepresentation: Data) throws -> Certificate.PrivateKey {
+        switch self {
+        case .cms_1_0_0:
+            // TODO: don't need this check after https://github.com/apple/swift-package-manager/pull/6138
+            #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+            if #available(macOS 11.0, iOS 14.0, watchOS 7.0, tvOS 14.0, *) {
```
Comment (yim-lee) [inline Sources/PackageSigning/SignatureProvider.swift:121]: @neonichu Do you know how far away we are from merging https://github.com/apple/swift-package-manager/pull/6138?

**Comment by yim-lee** [comment]
@swift-ci please smoke test macOS

**Review by neonichu — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci smoke test

**Comment by tomerd** [comment]
@swift-ci smoke test

**Comment by tomerd** [comment]
@swift-ci smoke test

**Comment by tomerd** [comment]
@swift-ci smoke test

**Comment by tomerd** [comment]
@swift-ci smoke test macOS

**Comment by tomerd** [comment]
@swift-ci smoke test macOS

**Comment by tomerd** [comment]
@swift-ci smoke test macOS

**Comment by tomerd** [comment]
@swift-ci smoke test macOS

**Comment by tomerd** [comment]
@swift-ci smoke test macOS

**Comment by tomerd** [comment]
@swift-ci smoke test macOS

**Comment by tomerd** [comment]
@swift-ci smoke test linux

**Comment by tomerd** [comment]
@yim-lee this should be ready now

**Comment by yim-lee** [comment]
@swift-ci please smoke test Windows

**Comment by yim-lee** [comment]
@swift-ci please smoke test Windows

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test Windows

**Comment by yim-lee** [comment]
Thanks @tomerd. Merging.

**Pull request merged**